tests/dataframe/test_creation.py

```diff
@@ -511,6 +511,86 @@ def test_create_dataframe_csv_specify_schema_no_headers(
+def test_create_dataframe_csv_specify_schema_no_headers_partial_schema(
+    valid_data: list[dict[str, float]], use_native_downloader
+) -> None:
+    with create_temp_filename() as fname:
+        with open(fname, "w") as f:
+            header = list(valid_data[0].keys())
+            writer = csv.writer(f, delimiter="\t")
+            writer.writerows([[item[col] for col in header] for item in valid_data])
+            f.flush()
+
+        with pytest.raises(ValueError):
+            daft.read_csv(
+                fname,
+                delimiter="\t",
+                schema_hints={
+                    "sepal_length": DataType.float64(),
+                    "sepal_width": DataType.float64(),
+                },
+                has_headers=False,
+                use_native_downloader=use_native_downloader,
+            )
+
+
+@pytest.mark.parametrize("use_native_downloader", [True, False])
+def test_create_dataframe_csv_schema_hints_partial(valid_data: list[dict[str, float]], use_native_downloader) -> None:
+    with create_temp_filename() as fname:
+        with open(fname, "w") as f:
+            header = list(valid_data[0].keys())
+            writer = csv.writer(f, delimiter="\t")
+            writer.writerow(header)
+            writer.writerows([[item[col] for col in header] for item in valid_data])
+            f.flush()
+
+        df = daft.read_csv(
+            fname,
+            delimiter="\t",
+            schema_hints={
+                "sepal_length": DataType.float64(),
+                "sepal_width": DataType.float64(),
+            },
+            use_native_downloader=use_native_downloader,
+        )
+        assert df.column_names == COL_NAMES
+
+        pd_df = df.to_pandas()
+        assert list(pd_df.columns) == COL_NAMES
+        assert len(pd_df) == len(valid_data)
+
+
+@pytest.mark.parametrize("use_native_downloader", [True, False])
+def test_create_dataframe_csv_schema_hints_override_types(
+    valid_data: list[dict[str, float]], use_native_downloader
+) -> None:
+    with create_temp_filename() as fname:
+        with open(fname, "w") as f:
+            header = list(valid_data[0].keys())
+            writer = csv.writer(f, delimiter="\t")
+            writer.writerow(header)
+            writer.writerows([[item[col] for col in header] for item in valid_data])
+            f.flush()
+
+        df = daft.read_csv(
+            fname,
+            delimiter="\t",
+            schema_hints={
+                "sepal_length": DataType.string(),  # Override the inferred float64 type to string
+            },
+            use_native_downloader=use_native_downloader,
+        )
+        assert df.column_names == COL_NAMES
+
+        pd_df = df.to_pandas()
+        assert list(pd_df.columns) == COL_NAMES
+        assert len(pd_df) == len(valid_data)
+
+        assert pd_df["sepal_length"].dtype == "object"
+        assert pd_df["sepal_length"][0] == str(valid_data[0]["sepal_length"])
+
+
```

Comment:
Another corner-case we should probably add is also to ensure that when we pass in some random schema hint (e.g. `"foo": ...`) our code correctly ignores it without any exceptions.

Reply:
added this test case as well!